### PR TITLE
My tickets page for pc finished 

### PR DESCRIPTION
**Pull request by littleCitizen6**

### CODE REVIEW

File: src/Components/CommmonComponents/Table/Table.js [hunk 1]
```diff
@@ -31,11 +48,89 @@ const useStyles = makeStyles({
     minWidth: 700,
   },
 });
+function TablePaginationActions(props) {
```
Comment (littlecitizen11) [inline src/Components/CommmonComponents/Table/Table.js:51]: think maybe change this table function to complete another component

File: src/Components/CommmonComponents/Table/Table.js [hunk 2]
```diff
@@ -31,11 +48,89 @@ const useStyles = makeStyles({
...
+    setPage(newPage);
+  };
+
+  const handleChangeRowsPerPage = (event) => {
+    setRowsPerPage(parseInt(event.target.value, 10));
+    setPage(0);
+  };
   const headKey = Object.keys(props.rows[0])[0];
 
+  const onClick = (rowId) =>{
```
Comment (littlecitizen11) [inline src/Components/CommmonComponents/Table/Table.js:127]: use a indecative name to this function. 
In addition, left a comment to make easy the next programmer who will use this table (like users table - on click route to edit user)

File: src/Components/CommmonComponents/Table/Table.js [hunk 3]
```diff
@@ -47,14 +142,36 @@ export default function CustomizedTable(props) {
           </TableRow>
         </TableHead>
         <TableBody>
-          {props.rows.map((row) => (
-            <StyledTableRow key={row[headKey]}>
+          {(rowsPerPage > 0
+            ? props.rows.slice(page * rowsPerPage, page * rowsPerPage + rowsPerPage)
+            : props.rows)
+        //   props.rows
```
Comment (littlecitizen11) [inline src/Components/CommmonComponents/Table/Table.js:148]: delete this comment

File: src/Components/CommmonComponents/Table/Table.js [hunk 4]
```diff
@@ -47,14 +142,36 @@ export default function CustomizedTable(props) {
...
                 {Object.keys(row).map((key) => (
                     <StyledTableCell align="right">{row[key]}</StyledTableCell>
                 ))}
             </StyledTableRow>
           ))}
         </TableBody>
+        <TableFooter>
+          <TableRow>
+            <TablePagination
+              rowsPerPageOptions={[5, 10, 25, { label: 'All', value: -1 }]}
```
Comment (littlecitizen11) [inline src/Components/CommmonComponents/Table/Table.js:160]: change this pagenation to a little config var to make it easier to change

File: src/Components/Pages/MyTicketsPage/MyTicketPageStyle.js [hunk 5]
```diff
@@ -0,0 +1,37 @@
+
+import { makeStyles } from '@material-ui/core/styles';
+
+
```
Comment (littlecitizen11) [inline src/Components/Pages/MyTicketsPage/MyTicketPageStyle.js:4]: love this PageStyle !

File: src/Components/Pages/MyTicketsPage/MyTicketsPage.js [hunk 6]
```diff
@@ -0,0 +1,179 @@
...
+    </Typography>
+  );
+}
+
+export default function MyTicketsPage(props) {
+  const classes = useStyles();
+  const [rows, setRows] = useState([
+      {loading:"loading..."}
+  ]);
+//   const [groups, setGroups] = useState([]);
```
Comment (littlecitizen11) [inline src/Components/Pages/MyTicketsPage/MyTicketsPage.js:37]: delete comment

File: src/Components/Pages/MyTicketsPage/MyTicketsPage.js [hunk 7]
```diff
@@ -0,0 +1,179 @@
...
+      {new Date().getFullYear()}
+      {' :כל הזכויות שמורות '}
+    </Typography>
+  );
+}
+
+export default function MyTicketsPage(props) {
+  const classes = useStyles();
+  const [rows, setRows] = useState([
+      {loading:"loading..."}
```
Comment (littlecitizen11) [inline src/Components/Pages/MyTicketsPage/MyTicketsPage.js:35]: where is it come in rendering? it wouldnt show

File: src/Components/Pages/MyTicketsPage/MyTicketsPage.js [hunk 8]
```diff
@@ -0,0 +1,179 @@
...
+    })
+    return translatedResults
+}
+
+function FormatDate(date){
+    let dateParts = date.split("T")[0].split("-");
+    return `${dateParts[2]}/${dateParts[1]}/${dateParts[0]}`
+}
+
+function FixedDateFormat(results){
```
Comment (littlecitizen11) [inline src/Components/Pages/MyTicketsPage/MyTicketsPage.js:69]: merge those 2 function into 2.. it not necessery to split them

File: src/Components/Pages/MyTicketsPage/MyTicketsPage.js [hunk 9]
```diff
@@ -0,0 +1,179 @@
...
+  useEffect(() => {
+    async function OnLoad() {
+      let groups = await getgroups();
+      let results = [];
+      groups.map(async group =>{
+          let costumerGroupRes = (await axios.get(`http://10.1.0.14:3000/Tickets/?costumerGroupId=${group.GroupId}`)).data;
+          let handlerGroupRes = (await axios.get(`http://10.1.0.14:3000/Tickets/?handlerGroupId=${group.GroupId}`)).data;
+          results.push(...costumerGroupRes,...handlerGroupRes);
+      }) 
+      let tickets = FixedDateFormat(results);
```
Comment (littlecitizen11) [inline src/Components/Pages/MyTicketsPage/MyTicketsPage.js:115]: why making those fixFunction ?
you may make this map function in the render return and choose only the .Name...
you wouldnt know either in the future someone will use the id or something....

File: src/Components/Pages/MyTicketsPage/MyTicketsPage.js [hunk 10]
```diff
@@ -0,0 +1,179 @@
...
+      tickets = FixedStatus(tickets, MapStatuses(statuses));
+      let allGroups = (await axios.get("http://10.1.0.19:3000/Groups/")).data;
+      tickets = FixedGroups(tickets, MapGroups(allGroups));
+      //add phones
+      let translatedData = TranslateKeys(tickets);
+      setRows(translatedData);
+    } OnLoad();
+   } , []);
+
+   const getgroups = async () =>{
```
Comment (littlecitizen11) [inline src/Components/Pages/MyTicketsPage/MyTicketsPage.js:126]: convention... getGroups..

File: src/Components/Pages/MyTicketsPage/MyTicketsPage.js [hunk 11]
```diff
@@ -0,0 +1,179 @@
...
+      let cookie = Cookies.get();
+      try {
+        let res = await axios.get(`http://10.1.0.19:3000/Groups/GroupByUsername/${cookie.username}`, { withCredentials: true });
```
Comment (littlecitizen11) [inline src/Components/Pages/MyTicketsPage/MyTicketsPage.js:129]: why not doing that in the useEffect as well ?

File: src/Components/Pages/MyTicketsPage/MyTicketsPage.js [hunk 12]
```diff
@@ -0,0 +1,179 @@
...
+      } 
+      catch (err) {
+       console.log("קיימת בעיה בשרת, אנא פנה לגורמים הרלוונטים"); 
+      }
+  
+
+   }
+  return (
+    <React.Fragment>
+      <CssBaseline />
```
Comment (littlecitizen11) [inline src/Components/Pages/MyTicketsPage/MyTicketsPage.js:140]: there is a problem with your backgrounds in the header and buttom component (they are white and the middle is other color)

File: src/Components/Pages/MyTicketsPage/MyTicketsPage.js [hunk 13]
```diff
@@ -0,0 +1,179 @@
...
+      {/* add nav-bar */}
+      <main>
+        <div className={classes.heroContent}>
+          <Container maxWidth="md">
+            <Typography component="h1" variant="h2" align="center" color="textPrimary" gutterBottom>
```
Comment (littlecitizen11) [inline src/Components/Pages/MyTicketsPage/MyTicketsPage.js:145]: there is a convention to a page (look at createTicket page for ex.) --> a logo and tinyer header

File: src/Components/Pages/MyTicketsPage/MyTicketsPage.js [hunk 14]
```diff
@@ -0,0 +1,179 @@
...
+              התקלות שלי
+            </Typography>
+            <div className={classes.heroButtons}>
+              <Grid container spacing={2} justify="flex-end">
+                <Grid item>
+
+                 <Fab color="primary" aria-label="Add" onClick={() =>history.push({pathname:'/CreateTicket'})} >
```
Comment (littlecitizen11) [inline src/Components/Pages/MyTicketsPage/MyTicketsPage.js:152]: the button looks like it there by mistake :P
make the button closer to the table...

File: src/Components/Pages/MyTicketsPage/MyTicketsPage.js [hunk 15]
```diff
@@ -0,0 +1,179 @@
...
+      <footer className={classes.footer}>
+        <Typography variant="h6" align="center" gutterBottom>
+          לבן היקר
+        </Typography>
+        <Typography variant="subtitle1" align="center" color="textSecondary" component="p">
+          אין לי מה להוסיף
+        </Typography>
+        <Copyright />
+      </footer>
+    </React.Fragment>
```
Comment (littlecitizen11) [inline src/Components/Pages/MyTicketsPage/MyTicketsPage.js:177]: why using react fragment?

File: src/Components/Pages/MyTicketsPage/MyTicketsPage.js [hunk 16]
```diff
@@ -0,0 +1,179 @@
...
+        </div>
+        <Container className={classes.cardGrid} maxWidth="lg">
+          <Grid container spacing={5}>
+              <Table rows={rows} setId={props.setTicketId} rowNavigation="/EditTicket">
+              </Table>
+          </Grid>
+        </Container>
+      </main>
+      
+      <footer className={classes.footer}>
```
Comment (littlecitizen11) [inline src/Components/Pages/MyTicketsPage/MyTicketsPage.js:168]: name convention footer...

File: src/Components/Pages/MyTicketsPage/MyTicketsPage.js [hunk 17]
```diff
@@ -0,0 +1,179 @@
...
+import Typography from '@material-ui/core/Typography';
+import Container from '@material-ui/core/Container';
+import Link from '@material-ui/core/Link';
+import Fab from '@material-ui/core/Fab';
+import AddIcon from '@material-ui/icons/Add';
+import Table from './../../CommmonComponents/Table/Table.js'
+import useStyles from './MyTicketPageStyle.js';
+import axios from 'axios';
+import { useHistory } from "react-router-dom";
+import { getgroups } from 'process';
```
Comment (littlecitizen11) [inline src/Components/Pages/MyTicketsPage/MyTicketsPage.js:13]: you dont use this..

**Review by littlecitizen11 — COMMENTED** [review]
;)

File: src/Components/CommmonComponents/Table/Table.js (hunk 1 above)
Comment (littleCitizen6) [reply src/Components/CommmonComponents/Table/Table.js:51]: fixed

File: src/Components/CommmonComponents/Table/Table.js (hunk 2 above)
Comment (littleCitizen6) [reply src/Components/CommmonComponents/Table/Table.js:127]: fixed

File: src/Components/CommmonComponents/Table/Table.js (hunk 3 above)
Comment (littleCitizen6) [reply src/Components/CommmonComponents/Table/Table.js:148]: fixed

File: src/Components/CommmonComponents/Table/Table.js (hunk 4 above)
Comment (littleCitizen6) [reply src/Components/CommmonComponents/Table/Table.js:160]: fixed

File: src/Components/Pages/MyTicketsPage/MyTicketsPage.js (hunk 17 above)
Comment (littleCitizen6) [reply src/Components/Pages/MyTicketsPage/MyTicketsPage.js:13]: fixed

File: src/Components/Pages/MyTicketsPage/MyTicketsPage.js (hunk 7 above)
Comment (littleCitizen6) [reply src/Components/Pages/MyTicketsPage/MyTicketsPage.js:35]: cause it is ready for the advenced filter case

File: src/Components/Pages/MyTicketsPage/MyTicketsPage.js (hunk 6 above)
Comment (littleCitizen6) [reply src/Components/Pages/MyTicketsPage/MyTicketsPage.js:37]: fixed

File: src/Components/Pages/MyTicketsPage/MyTicketsPage.js (hunk 8 above)
Comment (littleCitizen6) [reply src/Components/Pages/MyTicketsPage/MyTicketsPage.js:69]: i want to seperrate each formating cause for exemple, dar needed the datd formated except the translation so i seperated each one

File: src/Components/Pages/MyTicketsPage/MyTicketsPage.js (hunk 10 above)
Comment (littleCitizen6) [reply src/Components/Pages/MyTicketsPage/MyTicketsPage.js:126]: fixed

File: src/Components/Pages/MyTicketsPage/MyTicketsPage.js (hunk 13 above)
Comment (littleCitizen6) [reply src/Components/Pages/MyTicketsPage/MyTicketsPage.js:145]: this is only for forms' doesnt it?

File: src/Components/Pages/MyTicketsPage/MyTicketsPage.js (hunk 15 above)
Comment (littleCitizen6) [reply src/Components/Pages/MyTicketsPage/MyTicketsPage.js:177]: what is the footer name convension?

**Pull request merged**